### PR TITLE
Report dynamo expected-failure XPASSes

### LINKED ISSUE #139080: [dynamo] `dynamo_expected_failures` is silent on certain tests that ended up passing
### 🐛 Describe the bug

I discovered that some tests from `test/dynamo` are put in `dynamo_expected_failures`, yet those tests are passing with `PYTORCH_TEST_WITH_DYNAMO=1`, and we never saw any error to suggest removing those expected failures.

This suggests we might have more things in `dynamo_expected_failures` than we need.

For instance, see #138935.

### Versions

Python 3.12.5, de54246c422

### PR DESCRIPTION
**Pull request by jansel**
Stack from [ghstack](https://github.com/ezyang/ghstack/tree/0.16.0) (oldest at bottom):
* __->__ #186189

Direct `PYTORCH_TEST_WITH_DYNAMO=1` runs of tests under `test/dynamo`
were returning before invoking the common unittest runner because
`torch._dynamo.test_case.run_tests()` treated Dynamo mode the same as
crossref mode. That meant a test listed in `test/dynamo_expected_failures`
could pass without running the wrapper that reports the stale xfail.

There was a second way to hide the same signal: the dynamo expected-failure
wrapper uses `functools.wraps`, so wrapping a method that already had
`@unittest.expectedFailure` copied `__unittest_expecting_failure__` onto the
wrapper. unittest then classified the wrapper's "Unexpected success" error as
an expected failure.

Fix this at the harness level. Let `torch._dynamo.test_case.run_tests()` run in
Dynamo mode while preserving the crossref early return, and strip unittest's
expected-failure marker from the dynamo expected-failure wrapper. This keeps the
Dynamo xfail signal authoritative when a stale entry starts passing.

Fixes #139080
Generated by my agent

Test Plan:
- PYTORCH_TEST_WITH_DYNAMO=1 python test/dynamo/test_modules.py NNModuleTests.test_lazy_module2
- PYTORCH_TEST_WITH_DYNAMO=1 python test/dynamo/test_unspec.py UnspecTests.test_builtin_max_min
- python test/dynamo/test_unittest.py TestUnittest.test_dynamo_run_tests_reports_expected_failure_xpass TestUnittest.test_dynamo_expected_failure_ignores_unittest_xfail_marker
- python test/dynamo/test_unittest.py
- PYTORCH_TEST_WITH_DYNAMO=1 python test/dynamo/test_unittest.py TestUnittest.test_SkipTest
- lintrunner -a

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @kadeng @chauhang @amjames @jataylo @azahed98

### CODE REVIEW

**Comment by jansel** [comment]
@claude review these changes

**Comment by jansel** [comment]
My agent says CI was failing because the subprocess regression test imported torch from the raw checkout in CI, where torch/version.py has not been generated. I updated the helper to run from the parent process's imported torch root, refreshed the affected dynamo_expected_failures entries exposed now that test/dynamo runs under PYTORCH_TEST_WITH_DYNAMO=1, and resubmitted.

**Comment by jansel** [comment]
My agent says the remaining Dynamo wrapped CI failures were stale/incorrect test manifests exposed by the harness fix. I removed stale Dynamo expected-failure entries, added expected-failure entries for real Dynamo-mode failures, and moved unittest-XFAIL XPASS cases to dynamo_skips so wrapper CI passes while preserving the normal-mode expected-failure signal.

**Comment by jansel** [comment]
My agent says the latest Dynamo wrapped CI failures were more manifest drift exposed by the harness fix. I removed stale expected-failure entries for conv1d_symint_padding and lazy_module6 after confirming they pass under PYTORCH_TEST_WITH_DYNAMO=1, and added an expected-failure entry for device_mesh_get_local_rank, which still hits the DeviceMesh fake-class Dynamo error locally. The unrelated inductor HuggingFace failure was a CUDA OOM in google/gemma-2-2b, so I left it for rerun.

**Comment by jansel** [comment]
My agent says the latest Dynamo wrapped CI failures were additional manifest drift exposed by the harness fix. I added Dynamo expected-failure markers for device_mesh_init_skip_after_graph_break, data_dependent_evaluate_expr_graph_break, and lazy_module_kwargs after reproducing each failure under PYTORCH_TEST_WITH_DYNAMO=1 and confirming the corresponding normal-mode tests still pass. The TorchTitan torchcomms process-group failure is outside this Dynamo harness change and should be rerun or separately classified.

**Comment by jansel** [comment]
My agent says the latest Dynamo wrapped CI failures were more manifest drift exposed by the harness fix. I added Dynamo expected-failure markers for TestFakeDistributedP2P.test_compiled_batch_isend_irecv_mixed_graph and UnspecTests.test_mark_unbacked after reproducing their wrapped-mode failures and confirming normal mode passes, and removed the stale NNModuleTests.test_lazy_module_no_cls_to_become marker after confirming it now passes under PYTORCH_TEST_WITH_DYNAMO=1.

**Comment by jansel** [comment]
My agent says the latest Dynamo wrapped CI failures were additional manifest drift exposed by the harness fix. I added Dynamo expected-failure markers for TestFakeDistributedP2P.test_compiled_fire_and_forget_isend_graph, UnspecTests.test_mark_unbacked_channels_last, and NNModuleTests.test_nn_moduledict_contains after reproducing each wrapped-mode failure and confirming normal mode passes. I also normalized the existing zero-byte marker files so lintrunner no longer reports trailing-newline errors. The inductor TorchBench failures are unrelated environment/baseline failures from missing TorchBench dependencies such as torchmultimodal/opacus/segment_anything_fast.

**Comment by jansel** [comment]
My agent says the latest Dynamo wrapped failures were more manifest drift exposed by the harness fix. I added zero-byte Dynamo expected-failure markers for TestFakeDistributedP2P.test_compiled_irecv_graph, UnspecTests.test_mark_unbacked_hint_consistency, and MiscTests.test_cast_no_recompile_after_graph_break after confirming the unspec and misc wrapped failures locally and checking the CI log for the fake-distributed graph-count failure. The inductor_timm mobilenetv2_100 failure is an eager_two_runs_differ accuracy issue outside this Dynamo harness change, so I left it for the new CI run/rerun.

**Comment by jansel** [comment]
My agent says the latest Dynamo wrapped CI failures were more manifest drift exposed by the harness fix. I added zero-byte Dynamo expected-failure markers for TestFakeDistributedP2P.test_compiled_isend_graph, DynamicShapesCtxManagerTests.test__enter__exit_autocast_dynamic_shapes, MiscTests.test_cat_unbacked, and RematerializeACNodesPassTests.test_ac_rematerialize_with_no_annotations after reproducing the wrapped failures and confirming normal mode passes. I also removed the stale AotAutogradFallbackTests.test_aot_sequence_nr marker after confirming wrapped mode now passes. The inductor_timm mobilenetv2_100 failure looks unrelated: the log showed eager_two_runs_differ before Dynamo produced any graphs.

**Comment by jansel** [comment]
My agent says the latest Dynamo wrapped CI failures were more manifest drift exposed by the harness fix. I removed the stale ReproTests.test_add_sub_alpha_out expected-failure marker after confirming wrapped mode now passes, added expected-failure markers for MiscTests.test_catch_watchings1 and TestFakeDistributedP2P.test_compiled_p2p_interleave_graph after reproducing their wrapped-mode failures and confirming normal mode passes, and added a free-threaded-only Dynamo xfail for MiscTests.test_backend_match_guard_multi_threads based on the py3.14t CI log. The remaining inductor_timm mobilenetv2_100 failure still looks unrelated: it is eager_two_runs_differ before Dynamo produces any graphs.

**Comment by jansel** [comment]
My agent says the latest Dynamo wrapped CI failures were more manifest drift exposed by the harness fix. I added zero-byte expected-failure markers for DynamicShapesCtxManagerTests.test__enter__exit_autocast_graph_break_dynamic_shapes, DynamicShapesCtxManagerTests.test_functorch_get_dynamic_layer_stack_depth_dynamic_shapes, MiscTests.test_catch_watchings2, and NNModuleTests.test_simple_torch_function after reproducing their wrapped-mode failures and confirming normal mode passes. The functorch stack-depth case also needs a compiled_autograd_skips marker because compiled-autograd cleanup raises after the expected-failure wrapper has already converted the failure to a skip. The inductor_timm mobilenetv2_100 and tf_efficientnet_b0 failures look unrelated: both are training eager_two_runs_differ with compilation time 0.

**Comment by jansel** [comment]
My agent says the latest Dynamo wrapped CI failures were more manifest drift exposed by the harness fix. I added zero-byte expected-failure markers for DynamicShapesCtxManagerTests.test_autograd_profiler_dynamic_shapes, DynamicShapesCtxManagerTests.test_functorch_low_level_gb_False_dynamic_shapes, and NNModuleTests.test_torch_function_with_closure after reproducing each under PYTORCH_TEST_WITH_DYNAMO=1 and confirming normal mode passes. The inductor_timm tf_efficientnet_b0 failure is unrelated: it is a training eager_two_runs_differ with Compilation time 0 before Dynamo produces any graphs.

**Comment by jansel** [comment]
My agent says the latest Dynamo wrapped CI failures were more manifest drift exposed by the harness fix. I added zero-byte expected-failure markers for TestFakeDistributedP2PSubgroup.test_compiled_batch_isend_irecv_subgroup_uses_group_local_rank, NNModuleTests.test_torch_mangled_class_name, RematerializeACNodesPassTests.test_ac_rematerialize_with_rng_ops_raises_error, and MiscTests.test_cell_output1 after reproducing each wrapped-mode failure and confirming normal mode passes. The sccache failure is infra-only, and the inductor/titan failures are unrelated or already classified broken trunk.

**Comment by jansel** [comment]
My agent says the latest Dynamo wrapped CI failures were more manifest drift exposed by the harness fix. I added zero-byte expected-failure markers for TestFakeDistributedP2PSubgroup.test_compiled_irecv_subgroup_uses_group_local_rank, OptimizedModuleTest.test_attr_precedence, and DynamicShapesCtxManagerTests.test_disable_saved_tensors_hooks_graph_break_dynamic_shapes after reproducing each wrapped-mode failure and confirming the corresponding normal-mode tests pass. Dr. CI classified the remaining inductor failures as flaky or broken trunk.

**Comment by jansel** [comment]
My agent says the latest Dynamo wrapped CI failures were more manifest drift exposed by the harness fix. I added zero-byte expected-failure markers for TestFakeDistributedP2PSubgroup.test_compiled_isend_subgroup_uses_group_local_rank, ExportTests.test_cond_raise_user_error_on_missing_args, and FunctionTests.test_addcmul_ after confirming each normal-mode test passes and each fails under PYTORCH_TEST_WITH_DYNAMO=1 before the marker update. The new CI run is queued.

**Comment by jansel** [comment]
My agent says the latest Dynamo wrapped CI failures were addressed. I removed the stale ReproTests.test_addr_alpha_beta_out expected-failure marker after confirming wrapped mode now passes, added markers for UnspecTests.test_multiple_consecutive_random_calls_before_graph and FunctionTests.test_attrgetter after reproducing wrapped-mode failures and confirming normal mode passes, and fixed the expected-failure wrapper so self.subTest failures are handled without hiding future XPASSes. The attrgetter case also needs a compiled_autograd_skips marker because compiled-autograd cleanup aborts after the expected failure is converted to a skip.

**Comment by jansel** [comment]
@claude review these changes

**Comment by jansel** [comment]
My agent says the latest Dynamo wrapped CI failures were more manifest drift exposed by the harness fix. I added zero-byte expected-failure markers for ReproTests.test_as_strided_on_existing_view_banned and FunctionTests.test_broadcast_foreach_pow after reproducing their wrapped-mode failures and confirming normal mode passes, removed the stale MiscTests.test_cast_no_recompile_after_graph_break marker after confirming wrapped mode now passes, and added a short comment documenting the unittest subTest expected-failure handling.

**Comment by jansel** [comment]
My agent says the latest Dynamo wrapped CI failures were additional manifest drift exposed by the harness fix. I added zero-byte expected-failure markers for ReproTests.test_autograd_function_ctx_stash_no_vc_check, CtxManagerTests.test__enter__exit_autocast, DecoratorTests.test_allow_in_graph_inside_compile_gives_clear_error, LoggingTests.test_cudagraph_static_inputs, ExcTests.test_graph_break_log_generic_jump, FunctionTests.test_compare_constant_and_tensor, SubGraphTests.test_capi_call1, MinifierTestsCPU.test_after_dynamo_accuracy_error_cpu, RematerializeACNodesPassTests.test_ac_rematerialize_with_selective_checkpoint_policy, and ComptimeTests.test_get_local after reproducing the wrapped-mode failures and confirming normal mode passes. I also removed the stale SkipNonTensorTests.test_add_tensor1 marker after confirming wrapped mode now passes. The remaining distributed OSDC failure was a runner-container hook error before tests ran, so I treated it as infra/unrelated.

**Comment by jansel** [comment]
My agent says the latest Dynamo wrapped CI failures were more manifest drift exposed by the harness fix. I added zero-byte expected-failure markers for ReproTests.test_compilation_metrics_on_error, HigherOrderOpTests.test_cond_free_variable_in_both_branches, DecoratorTests.test_assume_constant_result_on_cached_property, LoggingTests.test_global_state_guard_logging, ExcTests.test_graph_break_source_attribution_on_stack, AutogradFunctionTests.test_aliasing_output, CtxManagerTests.test__enter__exit_autocast_graph_break, SubGraphTests.test_capi_call2, MinifierTestsCPU.test_after_dynamo_compile_error_cpu, ComptimeTests.test_print_bt, and TestExplainWithBackend.test_explain_with_backend after reproducing the wrapped-mode failures and confirming normal mode passes. I also removed the stale SkipNonTensorTests.test_add_tensor2 marker after confirming wrapped mode now passes.

**Comment by jansel** [comment]
My agent says the latest Dynamo wrapped CI failures were more manifest drift exposed by the harness fix. I added zero-byte expected-failure markers for the current wrapped-mode-only failures, removed stale markers for HigherOrderOpTests.test_cond_source_fn_stack and SkipNonTensorTests.test_add_tensor_dict after confirming wrapped mode now passes, and resubmitted. Dr. CI classified the remaining inductor_huggingface DebertaV2ForMaskedLM failure as flaky/unrelated.

**Comment by jansel** [comment]
My agent says I rebased on current main and submitted a new ghstack head. This refresh adds Dynamo expected-failure markers for the latest wrapped failures, removes stale `SkipNonTensorTests.test_add_tensor_list`, and narrows the Dynamo expected-failure/skip wrappers from `BaseException` to `Exception` with KeyboardInterrupt regression coverage. Targeted Dynamo wrapped and normal-mode tests passed; full `lintrunner -a` still trips an unrelated current-main Pyrefly stub drift in `torch/nn/modules/distance.py`, so I validated the changed files with Pyrefly skipped.

**Comment by jansel** [comment]
@claude review these changes

**Comment by jansel** [comment]
My agent says the latest Dynamo wrapped CI failures were another manifest refresh exposed by the harness fix. I added zero-byte expected-failure markers for 12 wrapped-mode-only failures after reproducing them locally and confirming normal-mode counterparts pass, and removed stale markers for CtxManagerTests.test_disable_saved_tensors_hooks, FunctionTests.test_default_dict_closure, and SkipNonTensorTests.test_do_not_skip_side_effects after confirming wrapped mode now passes. test/dynamo/test_unittest.py, git diff --check, and lintrunner -a passed.

**Comment by jansel** [comment]
My agent says the latest Dynamo wrapped CI failures were a 20-test manifest refresh exposed by the harness fix. The uploaded JUnit artifacts contained nine additional failures hidden behind the per-shard fail-fast summaries, so I added 17 zero-byte expected-failure markers for all reproduced wrapped-only failures and removed stale markers for FunctionTests.test_default_dict_lambda, HooksTests.test_functools_arg_vary, and SkipNonTensorTests.test_recursive_list after confirming they now pass in wrapped mode. All 20 normal-mode counterparts passed, test/dynamo/test_unittest.py passed, and lintrunner -a passed.

**Comment by jansel** [comment]
My agent says the latest Dynamo wrapped CI had 26 unique final test failures across 21 shards, including 10 hidden by the runner summary. I audited the raw logs, reproduced every case, added 23 expected-failure markers, removed the stale CtxManager marker, and added the appropriate Dynamo skip for the unittest-XFAIL XPASS. The bytecode debugger now skips before the outer wrapper because its interactive protocol intentionally raises KeyboardInterrupt, preserving the harness regression that process-control exceptions are not swallowed. All 26 wrapped cases now pass, the normal-mode counterparts were checked, and lintrunner -a passes.

**Comment by jansel** [comment]
My agent says the latest Dynamo wrapped CI had 22 unique test failures across 21 shards, including 10 hidden by the per-job summaries. I audited every raw log, added 19 expected-failure markers for reproduced wrapper-only failures, and removed three stale markers that now XPASS. All 22 targeted wrapped-mode cases and their normal-mode counterparts pass after the refresh, the harness regression suite passes, and lintrunner -a is clean.

**Comment by jansel** [comment]
My agent says I audited all 21 failing Dynamo-wrapped shards and their XML artifacts; they contained 22 unique failures, including 11 omitted from the Dr. CI summary. I reproduced all 22 in wrapped mode and verified every normal-mode counterpart passes. This update adds 19 expected-failure markers, removes three stale markers, and adds the compiled-autograd suppression needed for the context-manager cleanup case. The Dynamo unittest regressions and lintrunner both pass.

**Comment by jansel** [comment]
@claude review these changes

**Comment by jansel** [comment]
My agent says the latest Dynamo-wrapped CI contained 18 unique failures across 21 shards, including seven omitted from Dr. CI. I reproduced all 15 wrapped-only failures and their passing normal-mode counterparts, added expected-failure markers, removed stale ExportTests and NNModuleTests markers, and replaced the version-sensitive float/complex marker with a Python-aware Dynamo xfail so Python 3.14 passes without breaking Python 3.13 and older. The harness tests and lintrunner -a pass.

**Comment by jansel** [comment]
@claude review these changes

**Comment by jansel** [comment]
My agent says this PR is being closed as part of a cleanup of open agentic PRs to focus review bandwidth on a smaller set; the linked issue remains open.
